fvm/state/accounts.go

```diff
@@ -287,7 +310,7 @@ func (a *Accounts) SetCode(address flow.Address, code []byte) error {
 	}
 
 	var prevCode []byte
-	prevCode, err = a.ledger.Get(string(address.Bytes()), string(address.Bytes()), keyCode)
+	prevCode, err = a.ledger.Get(name, string(address.Bytes()), keyCode)
```

Comment:
The first two arguments, "controller" and "owner", are historical and soon will be merged into just one. 

Instead of using one of them to store the contract name, keep the first two values as is, but store under a different key instead, e.g. `fmt.Sprintf("%s.%s", keyCode, name)`.

This also needs to be changed for all the other `Get`/`Set` cases below